pxtlib/emitter/assembler.ts

```diff
@@ -323,18 +316,22 @@ namespace ts.pxtc.assembler {
             let m = /^(\d+)$/i.exec(s)
             if (m) v = parseInt(m[1], 10)
 
+            // stack-specific processing
+
             // more special characters to handle
             if (s.indexOf("@") >= 0) {
                 m = /^(\w+)@(-?\d+)$/.exec(s)
                 if (m) {
                     if (mul != 1)
                         this.directiveError(lf("multiplication not supported with saved stacks"));
                     if (this.stackpointers.hasOwnProperty(m[1]))
+                        // TODO: ARM-specific???
                         v = 4 * (this.stack - this.stackpointers[m[1]] + parseInt(m[2]))
                     else
                         this.directiveError(lf("saved stack not found"))
                 }
 
+                // TODO: ARM-specific???
```

Comment:
specific to CPUs which can only load 8 bit of constant data from an instruction at a time - I think this includes AVR - the @hi and @lo get the first and second byte of the address of label relative to the beginning of the assembly file